pkg/netceptor/conn.go

```diff
@@ -312,6 +317,12 @@ func (li *Listener) AcceptLoop(ctx context.Context) {
 			rAddr, ok := conn.RemoteAddr().(Addr)
 			if ok {
 				go MonitorUnreachable(li.pc, doneChan, rAddr, ccancel)
+			} else {
+				li.s.Logger.Debug("Remote address is not a Receptor address, skipping unreachable monitoring")
+				defer func() {
+					TestOnContextCancelDeferCalled.Store(true)
+					ccancel()
+				}()
```

Comment:
According to this:
https://github.com/golang/go/blob/27937289dc9fccf1f5513475145799087f39b964/src/context/context.go#L230

It is safe to call the cancel function simultaneously or multiple times. I believe we don't need the lock.

We can simply do:
```suggestion
			} else {
				li.s.Logger.Debug("Remote address is not a Receptor address, skipping unreachable monitoring")
			}
```

And some lines above:
```
cctx, ccancel := context.WithCancel(li.s.context)
defer ccancel()
```

What do you think? I find it simpler and easier to understand.